airbyte-integrations/connectors/destination-redshift-v2/src/main/kotlin/io/airbyte/integrations/destination/redshift_v2/write/load/RedshiftInsertBuffer.kt

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.redshift_v2.write.load
+
+import io.airbyte.cdk.load.data.AirbyteValue
+import io.airbyte.cdk.load.data.ArrayValue
+import io.airbyte.cdk.load.data.BooleanValue
+import io.airbyte.cdk.load.data.DateValue
+import io.airbyte.cdk.load.data.IntegerValue
+import io.airbyte.cdk.load.data.NullValue
+import io.airbyte.cdk.load.data.NumberValue
+import io.airbyte.cdk.load.data.ObjectValue
+import io.airbyte.cdk.load.data.StringValue
+import io.airbyte.cdk.load.data.TimeWithTimezoneValue
+import io.airbyte.cdk.load.data.TimeWithoutTimezoneValue
+import io.airbyte.cdk.load.data.TimestampWithTimezoneValue
+import io.airbyte.cdk.load.data.TimestampWithoutTimezoneValue
+import io.airbyte.cdk.load.schema.model.TableName
+import io.airbyte.cdk.load.util.serializeToString
+import io.github.oshai.kotlinlogging.KotlinLogging
+import java.sql.PreparedStatement
+import java.sql.Timestamp
+import java.sql.Types
+import javax.sql.DataSource
+
+private val log = KotlinLogging.logger {}
+
+/**
+ * Accumulates records and flushes to Redshift in batches. NOT a @Singleton - created per-stream by
+ * AggregateFactory.
+ */
+class RedshiftInsertBuffer(
+    private val tableName: TableName,
+    private val dataSource: DataSource,
+    private val flushLimit: Int = 1000,
```

Comment:
Harmless but ultimately unnecessary abstraction. Can just be in the Aggregate itself